CDLUC3/dmsp_frontend_prototype
# Added access tokens to GraphQL request headers, updated middleware to handle authorization

Issue: #55 

- Added JWT token to headers for graphql requests on both the client- and server-side components. Updated `lib/graphql/client.ts` and `lib/graphql/apollo-wrapper.tsx`
- Added an example client-side component called 'client-graphql-test' to show how apollo client hooks can work on client side using the 'apollo-wrapper'.
- Updated middleware.ts for authorization and redirection and updated middleware spec
- Added a 'get-token' endpoint to get the token from the cookie and added unit test
- Updated auth.ts to use 'jose' instead of 'jsonwebtoken' for token verification due to errors related to the edge runtime not support Node.js 'crypto' module.
- Added utils:
--  `utils/authLink.ts` - Creates 'authLink' for the creation of the client instances
--  `utils/cookiesUtil.ts` - Contains a `deleteCookie` method
--  `utils/getAuthTokenServer.ts` - Gets JWT token from cookie

**To Test authentication and authorization**:
- Run development branch of `dmsp_backend_prototype`. (Make sure the JWT secret is the same for backend as frontend)
- Run this frontend branch and go to `/signup`. Signup with a new user. 
- Go to `/login` and sign in with new user. Notice the new '`dmspt`' httpOnly cookie that was added
- Navigate to `/dmps/10.48321/D1SP4H`. There are currently errors getting data for the page, but you should be able to acces that page.
- Delete the '`dmspt`' cookie and try navigating to `/dmps/10.48321/D1SP4H`. You should be redirected to /login page

**To Test Token in Header for Client Side**:
- Navigate to `/client-graphql-test` and look at your Network tab and filter for graphql. You should see the token in the Authorization header
